### ISSUE TITLE
Upgrade to Python 3.5

### ISSUE THREAD

**Issue by toolness** [18F/calc#766]
I recently found a Python 3.5 feature that I wanted to use in CALC, but we're on 3.4.  Specifically, it's [PEP 448 - Additional Unpacking Generalizations](https://www.python.org/dev/peps/pep-0448/), which allows us to write things like this:

```
>>> dict(**{'x': 1}, y=2, **{'z': 3})
{'x': 1, 'y': 2, 'z': 3}
```

I think this could be particularly nice for easily combining a bunch of different Django template context-like dictionaries into a single context.

**Comment by jseppi** [comment]
Sweet. I think this is the current Python buildpack that cloud.gov is using: https://github.com/cloudfoundry/python-buildpack/releases/tag/v1.5.9

Just need to change `runtime.txt` for cloud.gov (and `.travis.yml` for Travis).

**Comment by toolness** [comment]
Well, 3.6 came out recently, and it's one of the biggest releases since 3.0 (includes 16 PEPs). It'd be nice if we could just leapfrog to that.

Also, 3.5 supports type annotations, which would let us start using [mypy](http://mypy-lang.org/) to lint our code, which could be nice.

**Comment by jseppi** [comment]
Yeah, 3.6 would be cool!

Just checked on GovCloud-cloud.gov. Current python buildpack is 1.5.13, which unfortunately doesn't have Python 3.6: https://github.com/cloudfoundry/python-buildpack/releases/tag/v1.5.13

**Comment by toolness** [comment]
Oh, this was fixed by #1219. Closing!
